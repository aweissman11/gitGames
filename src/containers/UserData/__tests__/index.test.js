import React from 'react';
import { shallow } from 'enzyme';

import { UserData } from '../index';

describe('UserData', () => {
  
  it('should match the snapshot', () => {
    const mockProps = {
      "userData": {},
      "loadingUser": {
        "message": "Getting User Data",
        "loadingUser": true
      },
      "hasErrored": {
        "hasErrored": false,
        "message": "nope"
      }
    }
    const wrapper = shallow(<UserData { ...mockProps } text='word' instances={90} />)
    expect(wrapper).toMatchSnapshot()
  })
  
  describe('usernameDisplay', () => {
    it('should tell the user to use the search', () => {
      const mockProps = {
        "userData": {},
        "loadingUser": {
          "message": "Getting User Data",
          "loadingUser": true
        },
        "hasErrored": {
          "hasErrored": false,
          "message": "nope"
        }
      }
      const wrapper = shallow(<UserData { ...mockProps } loginPage={true} />)
      expect(wrapper.instance().usernameDisplay()).toEqual(<span className="loading-span">Please use the search</span>)
    })

    it('should say user doesnt exist', () => {
      const mockProps = {
        "hasErrored": {
          "hasErrored": true,
          "message": "nope"
        }
      }

      const wrapper = shallow(<UserData { ...mockProps } />)
      expect(wrapper.instance().usernameDisplay()).toEqual(<span className="loading-span">User does not exist</span>)
    })

    it('should say getting data', () => {
      const mockProps = {
        "userData": {
          "avatarUrl": "https://avatars0.githubusercontent.com/u/36015215?v=4",
          "createdAt": "2018-01-31T21:48:32Z",
          "email": "",
          "login": "TCraig7",
          "name": "Tara Craig"
        },
        "loadingUser": {
          "message": "retrieving data",
          "loadingUser": true
        },
        "hasErrored": {
          "hasErrored": false,
          "message": "nope"
        }
      }

      const wrapper = shallow(<UserData { ...mockProps } />)
      expect(wrapper.find('.loading-span').text()).toEqual('getting user data<Loading />');
      
    })
    
    it('should show username', () => {
      const mockProps = {
        "userData": {
          "avatarUrl": "https://avatars0.githubusercontent.com/u/36015215?v=4",
          "createdAt": "2018-01-31T21:48:32Z",
          "email": "",
          "login": "TCraig7",
          "name": "Tara Craig"
        },
        "loadingUser": {
          "message": "User data retrieved",
          "loadingUser": false
        },
        "hasErrored": {
          "hasErrored": false,
          "message": "nope"
        }
      }
      
      const wrapper = shallow(<UserData { ...mockProps } />)
      expect(wrapper.find('span').last().text()).toEqual("TCraig7");
    })
  })
  
})







