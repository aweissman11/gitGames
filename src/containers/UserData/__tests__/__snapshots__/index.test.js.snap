// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserData should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserData
    hasErrored={
      Object {
        "hasErrored": false,
        "message": "nope",
      }
    }
    instances={90}
    loadingUser={
      Object {
        "loadingUser": true,
        "message": "Getting User Data",
      }
    }
    text="word"
    userData={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header-spinner-box"
      >
        <div
          className="loading-spinner-circle"
        >
          <Loading
            className="username-spinner"
            color="#b7b7b7"
            delay={0}
            height={25}
            type="spokes"
            width={25}
          />
        </div>
      </div>,
      "className": "spinner-user-data",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="loading-spinner-circle"
        >
          <Loading
            className="username-spinner"
            color="#b7b7b7"
            delay={0}
            height={25}
            type="spokes"
            width={25}
          />
        </div>,
        "className": "header-spinner-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Loading
            className="username-spinner"
            color="#b7b7b7"
            delay={0}
            height={25}
            type="spokes"
            width={25}
          />,
          "className": "loading-spinner-circle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "username-spinner",
            "color": "#b7b7b7",
            "delay": 0,
            "height": 25,
            "type": "spokes",
            "width": 25,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header-spinner-box"
        >
          <div
            className="loading-spinner-circle"
          >
            <Loading
              className="username-spinner"
              color="#b7b7b7"
              delay={0}
              height={25}
              type="spokes"
              width={25}
            />
          </div>
        </div>,
        "className": "spinner-user-data",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="loading-spinner-circle"
          >
            <Loading
              className="username-spinner"
              color="#b7b7b7"
              delay={0}
              height={25}
              type="spokes"
              width={25}
            />
          </div>,
          "className": "header-spinner-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Loading
              className="username-spinner"
              color="#b7b7b7"
              delay={0}
              height={25}
              type="spokes"
              width={25}
            />,
            "className": "loading-spinner-circle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "username-spinner",
              "color": "#b7b7b7",
              "delay": 0,
              "height": 25,
              "type": "spokes",
              "width": 25,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
